server/sonar-server/src/main/java/org/sonar/server/activity/db/ActivityDao.java

```diff
@@ -19,46 +19,33 @@
  */
 package org.sonar.server.activity.db;
 
-import org.sonar.api.utils.System2;
+import org.sonar.api.ServerComponent;
 import org.sonar.core.activity.db.ActivityDto;
 import org.sonar.core.activity.db.ActivityMapper;
+import org.sonar.core.persistence.DaoComponent;
 import org.sonar.core.persistence.DbSession;
-import org.sonar.server.db.BaseDao;
-import org.sonar.server.search.IndexDefinition;
+import org.sonar.core.persistence.MyBatis;
 
-import java.util.List;
+public class ActivityDao implements DaoComponent, ServerComponent {
```

Comment:
Currently DaoComponent do not extend BatchComponent nor ServerComponent, as some of them are still used by batch. Target goal is that all DAOs are ServerComponent.